src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.cs

```diff
@@ -0,0 +1,275 @@
+
+            return true;
+        }
+
+        private IVsTask QueueColorSchemeUpdate()
+        {
+            // Wait until things have settled down from the theme change, since we will potentially be changing theme colors.
+            return VsTaskLibraryHelper.CreateAndStartTask(VsTaskLibraryHelper.ServiceInstance, VsTaskRunContext.UIThreadBackgroundPriority, UpdateColorScheme);
+        }
+
+        private void UpdateColorScheme()
+        {
+            AssertIsForeground();
+
+            // Simply return if we were queued to run during shutdown or the user is in High Contrast mode.
+            if (_isDisposed || SystemParameters.HighContrast)
+            {
+                return;
+            }
+
+            // Set Foreground colors to default when possible
+            TryToDefaultThemeColors();
+
+            // The color scheme that is currently applied to the registry
+            var currentColorScheme = GetCurrentColorScheme();
+
+            // If this is a known theme then, use the users choosen option. For unknown themes we default to VS2017
+            // since themes would have been created with this as the expected base colors.
+            var configuredColorScheme = IsKnownTheme()
+                ? _settingsManager.GetValueOrDefault(ColorSchemeOptions.SettingKey, defaultValue: ColorSchemeOptions.Enhanced)
+                : ColorSchemeOptions.VisualStudio2017;
+
+            if (currentColorScheme == configuredColorScheme)
+            {
+                return;
+            }
+
+            // Update Default Colors
+            SetColorScheme(configuredColorScheme);
+
+            // Broadcast that system color settings have changed to force the ColorThemeService to reload colors.
+            NativeMethods.PostMessage(NativeMethods.HWND_BROADCAST, NativeMethods.WM_SYSCOLORCHANGE, wparam: IntPtr.Zero, lparam: IntPtr.Zero);
+        }
+
+        private void TryToDefaultThemeColors()
+        {
+            var themeId = GetThemeId();
+
+            // Do not change for unknown themes or if we've already tried to default colors.
+            if (!IsKnownTheme(themeId)
+                || HasThemeBeenDefaulted(themeId))
+            {
+                return;
+            }
+
+            // The previous method for applying color schemes updated the color of classifications to the expected theme color. 
+            // However, this new method relies on theme colors being the default, so classifications should use DefaultColor when
+            // possible.
+            if (_colorApplier.TrySetForegroundColorsToDefault(themeId))
+            {
+                SetThemeWasDefaulted(themeId);
+            }
+        }
+
+        private bool HasThemeBeenDefaulted(Guid themeId)
+        {
+            using var registryRoot = VSRegistry.RegistryRoot(_serviceProvider, __VsLocalRegistryType.RegType_Configuration, writable: true);
```

Comment:
Does this need to be writable?

Reply:
It does. The call to CreateSubKey requires writable to be true.